internal/server/proto/server.proto

```diff
@@ -3178,6 +3200,98 @@ message ReleaseExpanded {
   StatusReport latest_status_report = 2;
 }
 
+/********************************************************************
+* Trigger URLs
+********************************************************************/
+
+// The Trigger message is a representation of a URL that can be requested
+// for invoking specific lifecycle operations on a projects applications. These
+// trigger URLs are intended to be used in a CI system for easy configuration.
+// The user is expected to configure and generate the URL ahead of time, and
+// Waypoint will generate a trigger based on the configuration here and return
+// a URL to make requests with.
+message Trigger {
```

Comment:
Oops, writing the CLI implementation made me realize this message needs an operation reference for what the trigger will be executing. Will update this PR with that.

Reply:
Done